-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33648/22550

ERROR: Build errors found during clang-tidy run.
```
Calibration/TkAlCaRecoProducers/plugins/TrackDistanceValueMapProducer.cc:123:30: error: no matching function for call to 'min' [clang-diagnostic-error]
    std::copy(v_dR2.begin(), std::min(v_dR2.size(), v_dR2.begin() + nthClosestTrack_), std::back_inserter(reduced_vdR2));
                             ^
/cvmfs/cms-ib.cern.ch/nweek-02679/slc7_amd64_gcc900/external/gcc/9.3.0/lib/gcc/x86_64-unknown-linux-gnu/9.3.0/../../../../include/c++/9.3.0/bits/algorithmfwd.h:383:5: note: candidate template ignored: deduced conflicting types for parameter '_Tp' ('unsigned long' vs. '__gnu_cxx::__normal_iterator<float *, std::vector<float, std::allocator<float>>>')
--
gmake: *** [config/SCRAM/GMake/Makefile.coderules:128: code-checks] Error 2
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```

--- a/Calibration/TkAlCaRecoProducers/plugins/TrackDistanceValueMapProducer.cc
+++ b/Calibration/TkAlCaRecoProducers/plugins/TrackDistanceValueMapProducer.cc
@@ -120,7 +120,9 @@ void TrackDistanceValueMapProducer::produce(edm::StreamID streamID,
 
     // just copy the first nth
     std::vector<float> reduced_vdR2;
-    std::copy(v_dR2.begin(), v_dR2.begin() + nthClosestTrack_, std::back_inserter(reduced_vdR2));
+    std::copy(v_dR2.begin(),
+              v_dR2.begin() + std::min(v_dR2.size(), static_cast<size_t>(nthClosestTrack_)),
+              std::back_inserter(reduced_vdR2));
     v2_dR2.push_back(reduced_vdR2);
   }
 

